torch/csrc/distributed/rpc/tensorpipe_utils.h

```diff
@@ -43,14 +132,19 @@ struct TensorpipeReadBuffers {
 TORCH_API std::tuple<tensorpipe::Message, TensorpipeWriteBuffers>
 tensorpipeSerialize(
     Message&& rpcMessage,
-    std::vector<c10::DeviceIndex> devices = {});
+    std::vector<c10::DeviceIndex> devices = {},
+    const std::shared_ptr<FullDeviceContext>& =
+        std::make_shared<FullDeviceContext>(/* noCuda */ true));
 
 // Allocate the buffers that will hold the incoming data. They will be managed
 // by the returned holder, which must be kept alive until the asynchronous read
 // has finished. Pointers to these buffers will be stored in-place in the
 // TensorPipe message.
 TORCH_API TensorpipeReadBuffers
-tensorpipeAllocate(tensorpipe::Message& tpMessage);
+tensorpipeAllocate(
+    tensorpipe::Message& tpMessage,
+    const std::shared_ptr<FullDeviceContext>& ctx =
+        std::make_shared<FullDeviceContext>(/* noCuda */ true));
```

Comment:
Same here: const ref to the value rather than to the pointer is more idiomatic IMO.